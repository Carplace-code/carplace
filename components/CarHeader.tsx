import AddToWishlistModal from "./AddToWishlistModal";

interface CarHeaderProps {
  carId: string;
}

export default function CarHeader({ carId }: CarHeaderProps) {
  return (
    <div className="relative w-full overflow-hidden rounded-md shadow">
      <div className="h-64 w-full bg-gray-200" />
      <div className="absolute bottom-4 left-6 text-white">
        <h1 className="text-2xl font-bold">Mercedes-Benz C200 2023</h1>
        <p className="text-sm">Santiago, Chile â€¢ 18,000 km</p>
      </div>
      <div className="absolute right-6 bottom-4 flex items-center space-x-4">
        <div className="text-xl font-semibold text-white">CLP $35,990,000</div>
        <AddToWishlistModal versionId={carId} />
      </div>
    </div>
  );
}
