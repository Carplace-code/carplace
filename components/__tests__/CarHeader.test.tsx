import { render } from "@testing-library/react";
import { describe, it, vi } from "vitest";

import CarHeader from "../CarHeader";

// Mock AddToWishlistModal
vi.mock("../AddToWishlistModal", () => ({
  default: ({ versionId }: { versionId: string }) => <div data-testid="mock-wishlist">Mock Modal for {versionId}</div>,
}));

describe("CarHeader", () => {
  it("renders correctly and matches snapshot", () => {
    const { container } = render(<CarHeader carId="car_123" />);
    expect(container).toMatchSnapshot();
  });
});
