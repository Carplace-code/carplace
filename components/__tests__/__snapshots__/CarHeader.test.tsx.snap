// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarHeader > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="relative w-full overflow-hidden rounded-md shadow"
  >
    <div
      class="h-64 w-full bg-gray-200"
    />
    <div
      class="absolute bottom-4 left-6 text-white"
    >
      <h1
        class="text-2xl font-bold"
      >
        Mercedes-Benz C200 2023
      </h1>
      <p
        class="text-sm"
      >
        Santiago, Chile â€¢ 18,000 km
      </p>
    </div>
    <div
      class="absolute right-6 bottom-4 flex items-center space-x-4"
    >
      <div
        class="text-xl font-semibold text-white"
      >
        CLP $35,990,000
      </div>
      <div
        data-testid="mock-wishlist"
      >
        Mock Modal for 
        car_123
      </div>
    </div>
  </div>
</div>
`;
